firmware/src/low_voltage_helper.cpp

```diff
@@ -0,0 +1,53 @@
+#include "low_voltage_helper.h"
+#include <Romi32U4Buzzer.h>
+
+static bool lastLVFlag = false;
+static bool currLVFlag = false;
+static volatile unsigned long lvCount = 0;
+
+static const char lvTune[] PROGMEM = "!L8 V8 A<A A<A A<A A<A R1R1";
+
+static Romi32U4Buzzer bzr;
+
+void LowVoltageHelper::update(uint16_t currVoltageMV) {
+  lastLVFlag = currLVFlag;
+
+  if (currVoltageMV < kMinOperatingMV) {
+    // We are now in a low voltage mode
+    currLVFlag = true;
+
+    if (!lastLVFlag) {
+      // We transitioned from Non-LV -> LV
+      // Start the count
+      lvCount = 0;
+    }
+    else {
+      // We are still in LV mode
+      lvCount++;
+    }
+  }
+  else {
+    // We are NOT in LV mode
+    currLVFlag = false;
+
+    if (lastLVFlag) {
+      // We transitioned from LV -> Non-LV
+      lvCount = 0;
+    }
+  }
+}
```

Comment:
Do we need a de-bounce on the transition from LV -> Non-LV too?

Reply:
I think we might be ok without it? I can add it in, but I'm not sure how much value it'll add, since if we're in LV mode and we get a spurious non-LV transition, chances are high that we'll end up back in the LV state again anyway

Reply:
Eh I went ahead and added the debounce logic for both directions 😄  it's in the form of a state machine and should work 🤞

Reply:
@AustinShalit @jpokornyiii sanity check pls :)